src/org/rascalmpl/interpreter/TraversalEvaluator.java

```diff
@@ -431,10 +445,20 @@ private IValue traverseADTOnce(IValue subject, CaseBlockList casesOrRules,
 			  // this indicates that a nested rewrite has applied to replace a valid child with an invalid child
 			  throw new UnexpectedType(cons.getConstructorType().getFieldTypes(), tf.tupleType(args), (AbstractAST) null);
 		  }
-		  
+
+		  if (memoize) {
+			if (rcons == null) {
```

Comment:
turn into an assert?

Reply:
This is left-over debug code. I have removed it. rcons really cannot be null here.